make-release.sh

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
+# Release process automation script.
+# Used to create branch/tag and update VERSION files
+# and and trigger release by force pushing changes to the release branch
+
+# set to 1 to actually trigger changes in the release branch
+TRIGGER_RELEASE=0
+NOCOMMIT=0
+MAJOR_VERSION_BUMP=0
+
+usage ()
+{
+  echo "Usage: $0 --repo [GIT REPO TO EDIT] --version [VERSION TO RELEASE] [OPTIONS]"
+  echo "Options:"
+  echo "    --help, -h"
+  echo "        Show this message."
+  echo "    --no-commit, -n"
+  echo "        Do not commit changes."
+  echo "    --trigger-release, -t"
+  echo "        Push changes to remote repo, triggering release."
+  echo "Example: $0 --repo git@github.com:eclipse/che-subproject --version 'v3.2.1' --trigger-release"; echo
+}
+
+while [[ "$#" -gt 0 ]]; do
+  case $1 in
+    '-r'|'--repo') REPO="$2"; shift 1;;
+    '-v'|'--version') VERSION="$2"; shift 1;;
+    '-t'|'--trigger-release')
+      if [[ ! ${NOCOMMIT} -eq 0 ]]; then
+        echo "Only one of --trigger-release and --no-commit may be specified"
+        exit 1
+      fi
+      TRIGGER_RELEASE=1; shift 0;;
+    '-n'|'--no-commit')
```

Comment:
What is the purpose of the no-commit option?
Is it something like dry-run? If yes - let's revise if output if full to understand if the script works correctly, now:
![Screenshot_20200203_143834](https://user-images.githubusercontent.com/5887312/73653830-dfac4500-4692-11ea-8083-2030ddae2890.png)


Also, maybe it would be better to move to dry-run option which would print all operation it going to do with remotes but without performing it, like:
```
  Creating remote branch BRANCH_NAME (v3.2.x) for repo URL
  Bumping version on master to `v3.3.0-SNAPHOT`
  Creating PR ...
```

Reply:
That would make more sense, this is mostly from copying the script in from other repos. @nickboldt WDYT?

Reply:
`nocommit` was essentially a `dry-run` option so I could generate plugin registry content w/o pushing commits, in order to verify what was generated before pushing

For that particular repo, we should really have a verification test that runs all the bash scripts from build/script/* in the correct order to make sure that generated yaml is correct.

And in fact since this script has now been copied 4 times to 4 repos, we might want to consider centralizing it into an upstream Che repo. Maybe che-parent? or a new che-release repo? Would be nice to not have the same boilerplate script across 4 repos... as a variation will be needed for che-parent, che-docs, and che repos too in the near future.